### PR TITLE
fix: [CDS-57667]: changes to support any folder path instead of only …

### PR DESCRIPTION
**Pull request by achyuth-harness**
…charts/

## Describe your changes
Refactor the sub-charts folder path to any value besides just `charts/`

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45992)

### CODE REVIEW

File: 125-cd-nextgen/src/main/java/io/harness/cdng/K8sHelmCommonStepHelper.java [hunk 1]
```diff
@@ -975,14 +974,14 @@ public StepResponse handleHelmValuesFetchFailure(HelmValuesFetchResponsePassThro
   }
 
   public static List<HelmFetchFileConfig> mapHelmChartManifestsToHelmFetchFileConfig(
-      String identifier, List<String> valuesPaths, String manifestType, String subChartName) {
+      String identifier, List<String> valuesPaths, String manifestType, String subChartPath) {
     List<HelmFetchFileConfig> helmFetchFileConfigList = new ArrayList<>();
-    if (isEmpty(subChartName)) {
+    if (isEmpty(subChartPath)) {
```
Comment (abhinavsingh1196) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/K8sHelmCommonStepHelper.java:979]: This if-else block can be simplified to a single if

...
String valuesPath = VALUES_YAML_KEY;
if (isNotEmpty(subChartPath)) {
    valuesPath = Paths.get(subChartPath, VALUES_YAML_KEY).toString();
}

helmFetchFileConfigList.add(createHelmFetchFileConfig(identifier, manifestType, List.of(valuesPath), true));
...

**Review by abhinavsingh1196 — CHANGES_REQUESTED** [review]

**Review by achyuth-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45992)* status: 12 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/K8sHelmCommonStepHelper.java` line 979 at r3](https://reviewable.io/reviews/harness/harness-core/45992#-NS5EbXlxPHHLIWM6sYu-r3-979:-NS9rRfv6oU1dBXE5Ewu:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/56adc15eae85755c2956b86600ec5849ce2f2a85/125-cd-nextgen/src/main/java/io/harness/cdng/K8sHelmCommonStepHelper.java#L979)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

This if-else block can be simplified to a single if

...
String valuesPath = VALUES_YAML_KEY;
if (isNotEmpty(subChartPath)) {
    valuesPath = Paths.get(subChartPath, VALUES_YAML_KEY).toString();
}

helmFetchFileConfigList.add(createHelmFetchFileConfig(identifier, manifestType, List.of(valuesPath), true));
...
</blockquote></details>

Done.

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/k8sbase/K8sTaskHelperBase.java [hunk 2]
```diff
@@ -2923,7 +2923,7 @@ public List<FileData> renderTemplateForHelm(String helmPath, String manifestFile
             new HelmClientException(getErrorMessageIfProcessFailed("Failed to render template. ", processResult), USER,
                 HelmCliCommandType.RENDER_CHART));
       }
-      int index = isEmpty(subChartName)
+      int index = isEmpty(subChartPath)
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/k8sbase/K8sTaskHelperBase.java:2926]: Didn't we made this logic common? Not only for subchart?

**Review by achyuth-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45992)* status: 12 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/k8sbase/K8sTaskHelperBase.java` line 2926 at r4](https://reviewable.io/reviews/harness/harness-core/45992#-NS5EbXmU1fYd-iJxwlE-r4-2926:-NSAGEm_E9G1ebPB6LHN:bqj11z0) ([raw file](https://github.com/harness/harness-core/blob/6ef8a6a36f0fbe19b0ac9942a39aac8ec6b20a8e/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/k8sbase/K8sTaskHelperBase.java#L2926)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Didn't we made this logic common? Not only for subchart?
</blockquote></details>

Yes, we made this for native-helm flow (https://harness.atlassian.net/browse/CDS-53884), but need to do this here too. have updated

**Review by abosii-harness — APPROVED** [review]

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 2 of 8 files at r1, 4 of 6 files at r2, 3 of 4 files at r3, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45992)* status: 12 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]

**Pull request merged**